### PR TITLE
Comment Notifications: remove Trash swipe action

### LINKED ISSUE #18257: Unexpected result of Bin action in notification menu
### Expected behavior

Screenshot 2022-04-01 at 16 08 27

Tapping the big red Bin button, I expected to delete the notification not the actual comment I was notified about.

Even so, given that I am deleting other people's content I should have been prompted for confirmation. Ideally IMO I would just hide this power from the notification menu 😁 

### Actual behavior

I deleted the comment that I was notified about instead of the notification.

### Steps to reproduce the behavior

1. Get a notification on a P2 where you are editor
2. Swipe left on the notification
3. Tap bin
4. The comment or post gets deleted, right away

##### Tested on [device], iOS [version], WPiOS [version]
Latest iOS

### PR DESCRIPTION
**Pull request by ScoutHarris**
Fixes #18257 

This removes the comment `Trash` swipe action in the Notifications list. In addition:
- The word `Comment` has been added to the `Approve`/`Unapprove` button for clarity.
- Swipe action events are now tracked.
  - Approve/Unapprove - uses the existing `notifications_comment_(un)approved` events and adds a `notifications` or `notification_details` source.
  - Mark as read/unread - new `notification_mark_as_(un)read_tapped` events have been added.

| Before | After |
|--------|-------|
| unapprove_before | unapprove_after |
| approve_before | approve_after |

To test:

From Notifications > Comments list:
- Swipe left on a notification.
- Verify only the `Approve Comment`/`Unapprove Comment` button appears.
- Tap the button.
- Verify the events are now tracked:
`🔵 Tracked: notifications_comment_unapproved <blog_id: xxx, default_tab_experiment: nonexistent, site_type: p2, source: notifications>`
`🔵 Tracked: notifications_comment_approved <blog_id: xxx, default_tab_experiment: nonexistent, site_type: p2, source: notifications>`

From Notifications list:
- Swipe right on a notification.
- Select `Mark Unread`/`Mark Read`.
- Verify the events are now tracked:
`🔵 Tracked: notification_mark_as_unread_tapped <default_tab_experiment: nonexistent>`
`🔵 Tracked: notification_mark_as_read_tapped <default_tab_experiment: nonexistent>`

From Notifications > Comments list:
- Select a Comment notification.
- In the details, select `Pending` and `Approved`.
- Verify tracked events now include the `source`:
`🔵 Tracked: notifications_comment_unapproved <blog_id: xxx, default_tab_experiment: nonexistent, site_type: p2, source: notification_details>` 
`🔵 Tracked: notifications_comment_approved <blog_id: xxx, default_tab_experiment: nonexistent, site_type: p2, source: notification_details>`

## Regression Notes
1. Potential unintended areas of impact

2. What I did to test those areas of impact (or what existing automated tests I relied on)

3. What automated tests I added (or what prevented me from doing so)

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Comment by ScoutHarris** [comment]
Hey @mattmiklic . 

Just checking - are you ok with adding the word `Comment` to the `Approve`/`Unapprove` swipe buttons (as shown in the description)?

File: WordPress/Classes/Models/Notifications/Actions/ApproveComment.swift [hunk 1]
```diff
@@ -1,8 +1,8 @@
-/// Encapsulates logic to approve a cooment
+/// Encapsulates logic to approve a comment
```
Comment (dvdchr) [inline WordPress/Classes/Models/Notifications/Actions/ApproveComment.swift:1]: 👍!

**Review by dvdchr — APPROVED** [review]
Tested, works as described. :shipit:

**Comment by mattmiklic** [comment]
@ScoutHarris Yep, that seems to work fine.

**Pull request merged**